lib/home_page.dart

```diff
@@ -0,0 +1,326 @@
+  //    which might or might not have filter applied
+  void filterQuestionsCategory(value) {
+    selectedCategory = value;
+    if (value == "All Categories") {
+      questionsListFiltered = questionsListFilteredCategory = questionsListFilteredSearch;
+    } else {
+      setState(() {
+        questionsListFiltered = questionsListFilteredCategory = questionsListFilteredSearch
+            .where((question) =>
+                question.category.toLowerCase().contains(value.toLowerCase())
+                    ? true
+                    : false)
+            .toList();
+      });
+    }
+  }
+
+  // Main code for Home page
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      resizeToAvoidBottomPadding: false, // fixed: "Create New Question" button was moving up while in keyboard mode
+      appBar: PreferredSize(
+        preferredSize: Size.fromHeight(140.0), // here the desired height
+        child: AppBar(
+          // Menu button
+          leading: FlatButton(
+            textColor: Colors.white,
+            onPressed: () {}, // open side menu
+            child: Icon(Icons.menu, color: Colors.cyan),
+            shape: CircleBorder(side: BorderSide(color: Colors.transparent)),
+          ),
+
+          // Title "Eureka" is changed to a search bar after search icon is clicked
+          title: !isSearching
+              ? Text('Eureka!')
+              : TextField(
+                  onChanged: (value) {
+                    // Call function to apply search filter
+                    filterQuestionsSearch(value);
+                  },
+                  style: TextStyle(color: Colors.white),
+                  decoration: InputDecoration(
+                      icon: Icon(
+                        Icons.search,
+                        color: Colors.white,
+                      ),
+                      hintText: "Search Question",
+                      hintStyle: TextStyle(color: Colors.white)),
+                ),
+          centerTitle: true,
+
+          // Search icon is changed to search bar
+          actions: <Widget>[
+            isSearching
+                ? IconButton(
+                    icon: Icon(Icons.cancel),
+                    onPressed: () {
+                      searchingFalse();
+                    },
+                  )
+                : IconButton(
+                    icon: Icon(Icons.search),
+                    onPressed: () {
+                     searchingTrue();
+                    },
+                  ),
+
+            // Chat button
+            FlatButton(
+              textColor: Colors.white,
+              onPressed: () {}, // redirect to Chat page
+              child: Icon(Icons.chat_sharp, color: Colors.cyan),
+              shape: CircleBorder(side: BorderSide(color: Colors.transparent)),
+            ),
+          ],
+          backgroundColor: Colors.blueGrey[800],
+        ),
```

Comment:
Extract this outside of the `Widget build` something like this: 
```dart
AppBar homeAppBar() {
return AppBar(
          // Menu button
          leading: FlatButton(
            textColor: Colors.white,
            onPressed: () {}, // open side menu
            child: Icon(Icons.menu, color: Colors.cyan),
            shape: CircleBorder(side: BorderSide(color: Colors.transparent)),
          ),

          // Title "Eureka" is changed to a search bar after search icon is clicked
          title: !isSearching
              ? Text('Eureka!')
              : TextField(
                  onChanged: (value) {
                    // Call function to apply search filter
                    filterQuestionsSearch(value);
                  },
                  style: TextStyle(color: Colors.white),
                  decoration: InputDecoration(
                      icon: Icon(
                        Icons.search,
                        color: Colors.white,
                      ),
                      hintText: "Search Question",
                      hintStyle: TextStyle(color: Colors.white)),
                ),
          centerTitle: true,

          // Search icon is changed to search bar
          actions: <Widget>[
            isSearching
                ? IconButton(
                    icon: Icon(Icons.cancel),
                    onPressed: () {
                      searchingFalse();
                    },
                  )
                : IconButton(
                    icon: Icon(Icons.search),
                    onPressed: () {
                     searchingTrue();
                    },
                  ),

            // Chat button
            FlatButton(
              textColor: Colors.white,
              onPressed: () {}, // redirect to Chat page
              child: Icon(Icons.chat_sharp, color: Colors.cyan),
              shape: CircleBorder(side: BorderSide(color: Colors.transparent)),
            ),
          ],
          backgroundColor: Colors.blueGrey[800],
        );
}

Reply:
And then reference `homeAppBar()` instead. We don't want such bug chunks of code inside the `Widget build`.